src/test/java/seedu/address/model/person/PhoneTest.java

```diff
@@ -16,16 +16,22 @@ public void isValidPhone() {
         assertFalse(Phone.isValidPhone("phone")); // non-numeric
         assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
         assertFalse(Phone.isValidPhone("HIP: 9011041")); // Incorrect phone type
-        assertFalse(Phone.isValidPhone("HOP: 9011041 Oll: 9893898 Hm: 971897")); // Multiple incorrect phone type
+        assertFalse(Phone.isValidPhone("HOP: 9011041 Oll: 9893898 Hm: 971897")); // multiple incorrect phone type
+        assertFalse(Phone.isValidPhone("H: 911        ")); // consecutive whitespaces
+        assertFalse(Phone.isValidPhone("!@@#$#@$#@{}")); // random symbols
+
 
         // valid phone numbers
         assertTrue(Phone.isValidPhone("911")); // exactly 3 numbers
         assertTrue(Phone.isValidPhone("93121534"));
         assertTrue(Phone.isValidPhone("124293842033123")); // long phone numbers
-        assertTrue(Phone.isValidPhone("93121234 15341234")); // spaces within digits
+        assertTrue(Phone.isValidPhone("9312 981234 981534 17234")); // multiple numbers without labels
         assertTrue(Phone.isValidPhone("HP: 928782711")); // hand phone number
         assertTrue(Phone.isValidPhone("O: 9134343431")); // office number
         assertTrue(Phone.isValidPhone("H: 911")); // house number
-        assertTrue(Phone.isValidPhone("O: 911 H: 972727 HP: 9234817")); // office number
+        assertTrue(Phone.isValidPhone("O: 911 H: 972727 HP: 9234817")); // multiple numbers with different labels
+        assertTrue(Phone.isValidPhone("H: 916541 H: 916541 H: 965411 H: 965411")); // multiple numbers with same labels
+        assertTrue(Phone.isValidPhone("916541 H: 916541 965411 HP: 965411")); // mixture of numbers with and without labels
```

Comment:
I think allowing some phone numbers to be saved with labels and some without labels isn't good design. We'll have to discuss this further.

But I'm thinking of having fixed labels like home and mobile. And if the user only has one number and does not specify, we will assign it to maybe mobile by default. But only when the user specifies the labels then we will tokenise, trim the inputs and save them in a fixed format which is cleaner.

But I guess this would take some time to implement.